CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+cmake_minimum_required(VERSION 3.13.4)
+project(CMakePublic VERSION 1.0.0 LANGUAGES CXX)
+
+add_library(CMakePublic cmake_public.cpp)
+
+#include header file
+target_include_directories (CMakePublic PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
+
+install(TARGETS CMakePublic
+    EXPORT CMakePublic-targets
```

Comment:
I recommend using `${PROJECT_NAME}-targets` instead of `CMakePublic-targets` so you don't have to change this in the event that the project's name changes.